rct229/rulesets/ashrae9012019/section5/section5rule22.py

```diff
@@ -99,10 +100,11 @@ def get_calc_vals(self, context, data=None):
                     return {
                         "has_shading_overhang": subsurface_b["has_shading_overhang"],
                         "has_shading_sidefins": subsurface_b["has_shading_sidefins"],
+                        "depth_of_overhang": subsurface_b["depth_of_overhang"],
                     }
 
                 def rule_check(self, context, calc_vals=None, data=None):
-                    return not (
-                        calc_vals["has_shading_overhang"]
-                        or calc_vals["has_shading_sidefins"]
-                    )
+                    return (
+                        calc_vals["has_shading_overhang"] is False
+                        or calc_vals["depth_of_overhang"] == ZERO.LENGTH
+                    ) and calc_vals["has_shading_sidefins"] is False
```

Comment:
This is a style comment.
Usually we'd like to take out the variable in the `rule_check` function so that we can correct label the data source.
```python
has_shading_overhang_b = calc_vals["has_shading_overhang"]
depth_of_overhang_b = calc_vals["depth_of_overhang"]
has_shading_sidefins_b = calc_vals["has_shading_sidefins"]
return not has_shading_overhang_b and depth_of_overhang_b == ZERO.LENGTH and not has_shading_side_fins_b
```

Reply:
@claperle will you please confirm if you agree with my interpretation of this rule, "it seemed like it was a mistake in the way that the pseudo-python was written because the logic written in English does not match the pseudo-python logic"

I am suggesting changing the python logic to:
                        `(has_shading_overhang_b is False or depth_of_overhang_b == ZERO.LENGTH) and has_shading_sidefins_b is False`


while the RDS has written: if subsurface is flush with the exterior wall, and no shading projections are modeled or projection depth is 0. Then described in pseudo-python: `NOT (subsurface_b.has_shading_overhang OR subsurface_b.depth_of_overhang != 0)) AND ( NOT subsurface_b.has_shading_sidefins )`